### PR TITLE
DB Config.

### PR DESCRIPTION
**Pull request by Mvamd12**
In this update, I have taken the necessary steps to prepare the environment for integrating the database into our project. This includes adding the required configuration files and making several modifications to ensure seamless operation.

Key Changes:
Implemented a Docker configuration to facilitate easy deployment and maintain environment consistency across different development setups.
Created database, service, and schemas files that abstract database interactions, allowing for better organization of our code and separation of concerns.

Next Steps:
My partner and I will reformat the project and divide responsibilities to ensure efficient operation with the database.

### CODE REVIEW

**Review by Omarmanassrah — APPROVED** [review]

File: app/database.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker, declarative_base
+
+
+
+DATABASE_URL = "postgresql://postgres:password@db:5432/fastapi_db"
```
Comment (Mohammed-samaheen) [inline app/database.py]: This should be an environment variable.

File: app/schemas.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+from decimal import Decimal
+from app.api.routes.dependencies import get_current_time
+
+
+
+
+
+password_regex = r"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&#])[A-Za-z\d@$!%*?&#]{8,}$" 
+
+class Status:
```
Comment (Mohammed-samaheen) [inline app/schemas.py:15]: Remind me, why do we have this class?

File: app/schemas.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+class GetUserResponseModel(BaseModel):
+    id: UUID
+    username: str
+    email: EmailStr
+    is_admin: bool
+    is_active: bool
+    created_at: datetime
+    updated_at: datetime
+
+class Product:
```
Comment (Mohammed-samaheen) [inline app/schemas.py:111]: and this one

**Review by Mohammed-samaheen — CHANGES_REQUESTED** [review]

File: app/schemas.py (hunk 2 above)
Comment (Mvamd12) [reply app/schemas.py:15]: This is just what was approved in a previous PR for my partner, and as i mention in this PR details it's only for setting up the database and structural matters, my partner and me will distribute tasks to completely rebuild the project to work efficiently with the database. Please dont judge now.

File: app/schemas.py (hunk 3 above)
Comment (Mvamd12) [reply app/schemas.py:111]: ==

File: app/database.py (hunk 1 above)
Comment (Mvamd12) [reply app/database.py]: Done.

File: app/schemas.py (hunk 2 above)
Comment (Mohammed-samaheen) [reply app/schemas.py:15]: It was a question to see if we still need it or not. In that case, just remind @Omarmanassrah to take care of this part in his changes.

**Review by Mohammed-samaheen — APPROVED** [review]
Something to consider:
- I would rather use Pydantic settings to include environment variables.
-Add the required environment variables to the README file or set them as default values in the Docker Compose file.

**Pull request merged**